### PR TITLE
draft connector marketplace

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Closes https://github.com/camunda/web-modeler/issues/6460.

## To do

- [x] Distinctions between connectors that rely on our connector bundle runtime and those possessing their own runtime.
- [x] Add screenshots.
- [x] Procedures detailing how users can duplicate and manipulate a template from the marketplace.
- [x] Give feedback on new user experience.
- [x] Add to sidebar.
- [x] See if we need this documented for Self-Managed.
- [x] See if we need the actual browser Marketplace documented.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by christinaausley** [comment]
@bastiankoerber you can note my progress here, but I have a few questions before I'll be content with this documentation (as noted in the to do section of the PR description).

> Distinctions between connectors that rely on our connector bundle runtime and those possessing their own runtime.

**Can you please clarify this for me, or point me towards documentation so I can have a clear understanding of this?**

> Procedures detailing how users can duplicate and manipulate a template from the marketplace.

**What do you want to clarify here? The following screenshot?**

Screenshot 2023-10-25 at 3 01 52 PM

> Give feedback on new user experience.

**After stepping through this as a new user, I just have one piece of feedback. When I download the Connector, the actual "downloading" modals that pop up show up and disappear very quickly. Can we keep these up for the user to read, and just click to Continue? It felt like I was missing information (no matter how small) along the way.**
 
> Add to sidebar.

**Where would you like to see this placed? Within the Connectors docs, or within the Modeler docs? Regardless, we can link across.**

> See if we need this documented for Self-Managed. See if we need the actual browser Marketplace documented.

**Is this content you feel we are missing, or out of scope for this particular doc?**

**Comment by christinaausley** [comment]
Note that the approved final draft will be added to `8.3`.

**Comment by bastiankoerber** [comment]
> @bastiankoerber you can note my progress here, but I have a few questions before I'll be content with this documentation and a final review (as noted in the to do section of the PR description).
> 
> > Distinctions between connectors that rely on our connector bundle runtime and those possessing their own runtime.
> 
> **Can you please clarify this for me, or point me towards documentation so I can have a clear understanding of this?**
> 
HI @christinaausley,
Connectors who are using our connector bundle runtime, are connector templates who are reusing the camunda connectors (Protocol Connector Approach https://camunda.com/blog/2023/04/create-custom-template-based-on-rest-connector/, https://camunda.com/blog/2023/06/architecture-connectors/, https://www.youtube.com/watch?v=OCP3pXyIP_I, ) 

In Fact, it means, that the developer only created a template and reused one of our connector runtimes.
Important: Only for those templates the direct download from our marketplace is possible.

> > Procedures detailing how users can duplicate and manipulate a template from the marketplace.
> 
> **What do you want to clarify here? The following screenshot?**

Yes, this screenshot and the template editor.
For this screeshot it would be important to describe the actions:
- Save as => you are not overwriting the current connector, you save it as new file (which you can edit later on)
- replace => you are replacing the current connector (read only)
If you are downloading a connector from our marketplace, it is read-only. (You can see it, if you are opening the template in our template editor:
image
Here you need to click on customize template, to duplicate this template and be able to edit it.

> 
> Screenshot 2023-10-25 at 3 01 52 PM
> > Give feedback on new user experience.
> 
> **After stepping through this as a new user, I just have one piece of feedback. When I download the Connector, the actual "downloading" modals that pop up show up and disappear very quickly. Can we keep these up for the user to read, and just click to Continue? It felt like I was missing information (no matter how small) along the way.**
> 
> > Add to sidebar.
> 
> **Where would you like to see this placed? Within the Connectors docs, or within the Modeler docs? Regardless, we can link across.**

Good Question: We will extend the marketplace integration with other ressources in the near future (download process templates, forms DMN's etc.)
So i think a section in the web modeler would make sense. WDYT? But we should link it as well in our connectors documentation

> 
> > See if we need this documented for Self-Managed. See if we need the actual browser Marketplace documented.
> 
> **Is this content you feel we are missing, or out of scope for this particular doc?**

yes :-) Thank you

**Comment by christinaausley** [comment]
Hi all -- this is ready for final review. I'll backport once approved 👍

**Comment by akeller** [comment]
I saw @bastiankoerber is OOO and added @sbuettner to keep this moving.

**Comment by barmac** [comment]
Hi, I am adding @wollefitz as the reviewer since he's substituting @CatalinaMoisuc who's off till end of November.

File: docs/components/connectors/introduction.md
Comment (sbuettner) [inline docs/components/connectors/introduction.md:29]: I am not sure whether this categorization is really clear as the OOTB connectors come with Connector templates as well.

File: docs/components/connectors/use-connectors/index.md
Comment (sbuettner) [inline docs/components/connectors/use-connectors/index.md:9]:
```suggestion
Find the available Connectors in Camunda SaaS and how to use them in detail in the [out-of-the-box Connectors](/components/connectors/out-of-the-box-connectors/available-connectors-overview.md) documentation. Additionally, learn how you can visit the [Connector Marketplace](/docs/components/modeler/web-modeler/connector-marketplace.md) from your BPMN diagram.
```

File: docs/components/modeler/web-modeler/connector-marketplace.md
Comment (sbuettner) [inline docs/components/modeler/web-modeler/connector-marketplace.md:7]:
```suggestion
While modeling a process, you can click into the Connector Marketplace to browse through Connectors provided by Camunda engineering, discover Connectors developed by Camunda partners, and explore Connectors developed by the Camunda Community.
```

File: docs/components/modeler/web-modeler/connector-marketplace.md
Comment (sbuettner) [inline docs/components/modeler/web-modeler/connector-marketplace.md:29]:
```suggestion
Once you find a Connector you want to integrate into your BPMN diagram, click **Download to project**. The resource is then downloaded from the Connector Marketplace into your project.
```

**Review by sbuettner — COMMENTED** [review]
Thanks @christinaausley. Looks really good. Just some minor comments. I also currently dont know whether we use Camunda 8 just Camunda. Hence the comment regarding the product naming.

**Comment by akeller** [comment]
> I also currently dont know whether we use Camunda 8 just Camunda. Hence the comment regarding the product naming.

@sbuettner we'll continue to qualify the product with a number when we are in the docs. "Camunda 8" is the way to go (at least for now 😅 ).

File: docs/components/connectors/introduction.md
Comment (christinaausley) [reply docs/components/connectors/introduction.md:29]: Reworded in latest commit to:

Connectors are available [out-of-the-box (OOTB)](./out-of-the-box-connectors/available-connectors-overview.md) and come with [Connector Templates](manage-connector-templates.md) which customize how a BPMN element is shown,
and how it can be configured by process developers. Connector templates are a specific kind of [element templates](/components/modeler/desktop-modeler/element-templates/about-templates.md), which can also be used when creating custom Connectors via the [Connector SDK](./custom-built-connectors/connector-sdk.md).

WDYT?

File: docs/components/modeler/web-modeler/connector-marketplace.md
Comment (christinaausley) [reply docs/components/modeler/web-modeler/connector-marketplace.md:7]: Adjusted in latest commit 👍

File: docs/components/modeler/web-modeler/connector-marketplace.md
Comment (christinaausley) [reply docs/components/modeler/web-modeler/connector-marketplace.md:9]: Adjusted in latest commit 👍

File: docs/components/modeler/web-modeler/connector-marketplace.md
Comment (christinaausley) [reply docs/components/modeler/web-modeler/connector-marketplace.md:29]: Adjusted in latest commit 👍

**Comment by christinaausley** [comment]
@akeller Do you know if this should be backported to 8.3 or farther?

**Comment by akeller** [comment]
> @akeller Do you know if this should be backported just to 8.3 or farther?

@christinaausley I thought I saw a filter for versions including 8.2, but now I don't see it. If you want to merge this quickly, let's just backport to 8.3.

File: docs/components/modeler/web-modeler/img/connector-marketplace.png
Comment (akeller) [inline docs/components/modeler/web-modeler/img/connector-marketplace.png:1]: My filters look a bit different. I have the following:

* Categories
* Connector Type
* Creator
* Connector Runtime
* Supported BPMN Types

Nothing about the compatible version.

**Comment by wollefitz** [comment]
> Hi, I am adding @wollefitz as the reviewer since he's substituting @CatalinaMoisuc who's off till the end of November.

I'm passing the torch to @jfriedenstab as he's the DRI of most connector marketplace topics within the Web Modeler team and can give more valuable feedback.

File: docs/components/modeler/web-modeler/img/connector-marketplace.png
Comment (christinaausley) [reply docs/components/modeler/web-modeler/img/connector-marketplace.png:1]: Screenshot 2023-10-31 at 12 37 03 PM

@akeller This is what I am seeing today, and the same that @bastiankoerber shared. What environment are you running this in? I'm in `https://modeler.cloud.ultrawombat.com/`.

**Comment by christinaausley** [comment]
> > @akeller Do you know if this should be backported just to 8.3 or farther?
> 
> @christinaausley I thought I saw a filter for versions including 8.2, but now I don't see it. If you want to merge this quickly, let's just backport to 8.3.

Screenshot 2023-10-31 at 12 41 47 PM

hmmmmm 🐟

File: docs/components/modeler/web-modeler/img/connector-marketplace.png
Comment (akeller) [reply docs/components/modeler/web-modeler/img/connector-marketplace.png:1]: https://marketplace.camunda.com/en-US/listing

**Comment by akeller** [comment]
@christinaausley please ping on Slack in the ask-connectors channel and tag the remaining reviewers. I think we wanted to move quickly on this. I'm not sure if we are documenting the current status or what's in the ultrawombat environment.

File: docs/components/modeler/web-modeler/img/connector-marketplace.png
Comment (christinaausley) [reply docs/components/modeler/web-modeler/img/connector-marketplace.png:1]: I see. It looks like the .com view differs from the one you see in Modeler. I can add a note to clarify this.

**Comment by christinaausley** [comment]
> @christinaausley please ping on Slack in the ask-connectors channel and tag the remaining reviewers. I think we wanted to move quickly on this. I'm not sure if we are documenting the current status or what's in the ultrawombat environment.

I added a note to clarify this based on Bastian's response in [Slack](https://camunda.slack.com/archives/C02JLRNQQ05/p1698772218887819?thread_ts=1698771748.930359&cid=C02JLRNQQ05). This has also been backported and is ready for final review 👍

**Review by akeller — DISMISSED** [review]
This is a bit nit picky and IMO we don't need to address it here in this PR, but is it the "Connector Marketplace" or the "Camunda Marketplace"?

**Comment by bastiankoerber** [comment]
> This is a bit nit picky and IMO we don't need to address it here in this PR, but is it the "Connector Marketplace" or the "Camunda Marketplace"?

it should be Camunda Marketplace

**Comment by christinaausley** [comment]
Thank you! I'll adjust across this PR and then we should be clear to merge.

**Comment by christinaausley** [comment]
@bastiankoerber, @jfriedenstab, @sbuettner can I get a final review here? 😄

**Review by bastiankoerber — APPROVED** [review]

**Comment by jfriedenstab** [comment]
Sorry, I'm a bit late to the party 😊.
I noticed that the changes were also backported to 8.2, although the Marketplace integration in Web Modeler (Self-Managed) was only introduced with the 8.3 release. Maybe we should remove the references to the Marketplace from the 8.2 version to avoid confusion?

/cc @bastiankoerber

**Comment by bastiankoerber** [comment]
@jfriedenstab good catch. makes sense @christinaausley

**Pull request merged**
